superbench/benchmarks/micro_benchmarks/directx_gpu_encoding_latency.py

```diff
@@ -0,0 +1,157 @@
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+        self._parser.add_argument(
+            '--algo',
+            type=str,
+            choices=['ASAP', 'OneInOne'],
+            default='ASAP',
+            required=False,
+            help='The algorithm to use for encoding'
+        )
+        self._parser.add_argument(
+            '--codec',
+            type=str,
+            choices=['AVC', 'H264', 'HEVC', 'H265', 'AV1'],
+            default='H265',
+            required=False,
+            help='The codec to use for encoding'
+        )
+        self._parser.add_argument(
+            '--format',
+            type=str,
+            choices=['RGBA_F16', 'R10G10B10A2', 'NV12', 'P010'],
+            default='NV12',
+            required=False,
+            help='The format to use for encoding'
+        )
+        self._parser.add_argument(
+            '--frames', type=int, default=500, required=False, help='The number of frames to encode'
+        )
+        self._parser.add_argument(
+            '--height', type=int, default=720, required=False, help='The height of the input video'
+        )
+        self._parser.add_argument(
+            '--width', type=int, default=1080, required=False, help='The width of the input video'
+        )
+        self._parser.add_argument('--input_file', type=str, default=None, required=False, help='The input video file')
+        self._parser.add_argument('--output_file', type=str, default=None, required=False, help='The output video file')
+        self._parser.add_argument(
+            '--output_height', type=int, default=720, required=False, help='The height of the output video'
+        )
+        self._parser.add_argument(
+            '--output_width', type=int, default=1080, required=False, help='The width of the output video'
+        )
+        self._parser.add_argument(
+            '--vcn', type=int, choices=[0, 1], default=0, required=False, help='The VCN instance to use for encoding'
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        command = os.path.join(self._args.bin_dir, self._bin_name)
+        command += f' -ALGORITHM {self._args.algo}'
+        command += f' -CODEC {self._args.codec}'
+        command += f' -FORMAT {self._args.format}'
+        command += f' -FRAMES {self._args.frames}'
+        command += f' -HEIGHT {self._args.height}'
+        command += f' -WIDTH {self._args.width}'
+        if self._args.input_file is not None:
+            command += f' -INPUT {self._args.input_file}'
+        else:
+            if not os.path.exists(f'{self._test_file}.nv12'):
+                create_nv12_file(self._test_file, self._args.frames, self._args.width, self._args.height)
+            command += f' -INPUT {self._test_file}.nv12'
```

Comment:
strange here. self._test_file directly is better

Reply:
ok, updated